flight/Libraries/math/cic.c

```diff
@@ -0,0 +1,160 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsLibraries Tau Labs Libraries
+ * @{
+ * @addtogroup TauLabsMath Tau Labs math support libraries
+ * @{
+ *
+ * @file       pid.c
+ * @author     The OpenPilot Team, http://www.openpilot.org Copyright (C) 2010.
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2012-2014
+ * @brief      PID Control algorithms
+ *
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include <math.h>
+#include "misc_math.h"
+#include "cic.h"
+
+
+//! Store the filter parameter
+static uint8_t cic_filter_order     = 1;
```

Comment:
I also thought about, but was not sure if we need a cic filter somewhere else.
So i implemented it like it is done for pid.c/pid.h
So we just have one CIC filter setup, and can't use different instances. But otherwise we would store the CIC setup for all 3 axes, so some overhead.

And for SmallF1 targets cic.h will not be included, so we don't lose RAM with that on these targets.

But if you like to have more modular, i will change this.